### PR TITLE
Add hosting model switch tests

### PR DESCRIPTION
**Pull request by pakrym**
New test doesn't repro https://github.com/aspnet/IISIntegration/issues/1196

Also:

1. Made NativeSample support outofproc if we need to change hosting model for debugging
2. Fixed throwing and catching of exceptions in GlobalVerstionUtility 
3. Cleaned-up eventlog usages

### CODE REVIEW

File: build/launchSettings.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
       "nativeDebugging": true,
       "environmentVariables": {
         "IIS_SITE_PATH": "$(MSBuildThisFileDirectory)",
-        "ANCM_PATH": "$(AncmPath)",
-        "ANCMV2_PATH": "$(AncmV2Path)",
+        "ANCM_PATH": "$(TargetDir)\\$(AncmPath)",
```
Comment (pakrym) [inline build/launchSettings.json]: This change is to use applocal aspnetcorev2.dll and outofproc handler next to it.

File: samples/NativeIISSample/Startup.cs [hunk 2]
```diff
@@ -14,8 +14,15 @@ namespace NativeIISSample
 {
     public class Startup
     {
+        private readonly IAuthenticationSchemeProvider _authSchemeProvider;
+
+        public Startup(IAuthenticationSchemeProvider authSchemeProvider = null)
+        {
+            _authSchemeProvider = authSchemeProvider;
```
Comment (jkotalik) [inline samples/NativeIISSample/Startup.cs]: Nit: why move the AuthSchemeProvider out of Configure?

File: src/AspNetCoreModuleV2/AspNetCore/applicationmanager.cpp [hunk 3]
```diff
@@ -189,12 +189,6 @@ APPLICATION_MANAGER::RecycleApplicationFromManager(
 
         // Removed the applications which are impacted by the configurtion change
         m_pApplicationInfoHash->DeleteIf(FindConfigChangedApplication, (PVOID)&context);
-
-        if (m_pApplicationInfoHash->Count() == 0 && m_hostingModel == HOSTING_OUT_PROCESS)
-        {
-            // reuse current process
-            m_hostingModel = HOSTING_UNKNOWN;
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/AspNetCore/applicationmanager.cpp]: Why remove this check? What happens when we run an out of process app, shut it down, then start an inprocess app?

Comment (pakrym) [reply src/AspNetCoreModuleV2/AspNetCore/applicationmanager.cpp]: m_hostingModel  value never used. The logic you are talking about is in HandlerResolver

File: test/Common.FunctionalTests/ConfigurationChangeTests.cs [hunk 4]
```diff
@@ -54,5 +55,24 @@ public async Task ConfigurationChangeForcesChildProcessRestart(AncmVersion versi
             // Verify that worker process gets restarted with new process id
             await deploymentResult.HttpClient.RetryRequestAsync("/ProcessId", async r => await r.Content.ReadAsStringAsync() != processBefore);
         }
+
+        [ConditionalFact]
+        public async Task OutOfProcessToInProcessHostingModelSwitchWorks()
```
Comment (jkotalik) [inline test/Common.FunctionalTests/ConfigurationChangeTests.cs]: What about inprocess to out of process and expect a new process to start.

File: samples/NativeIISSample/Startup.cs (hunk 2 above)
Comment (pakrym) [reply samples/NativeIISSample/Startup.cs]: It's not available in outofproc so I've made it optional

File: test/Common.FunctionalTests/ConfigurationChangeTests.cs (hunk 4 above)
Comment (pakrym) [reply test/Common.FunctionalTests/ConfigurationChangeTests.cs]: It won't start our testsites do not support switching hosting mode.

**Pull request merged**